test.sh

```diff
@@ -0,0 +1,120 @@
+	if find . -mindepth 1 -maxdepth 1 ! -name "*.tex" -delete ; then
+		echo -e "${GREEN}OK${NOCOLOR}"
+	else
+		echo -e "${RED}FAIL${NOCOLOR}"
+		exit 1
+	fi
+}
+
+## This function runs pdflatex for a given file.
+## It generates a log file and in case of an error shows the problem in the console.
+## parameter $1 the name of the tex file (without .tex ending)
+## parameter $2 the number of the run (pdflatex requires multiple runs)
+run_pdflatex () {
+	echo -e -n "${GREEN}run $2 pdflatex for $1...${NOCOLOR}"
+	if pdflatex $3 -halt-on-error $1 2>&1 > pdflatex_$1_$2.log ; then
+		echo -e "${GREEN}OK${NOCOLOR}"
+	else
+		echo -e "${RED}fail${NOCOLOR}" >&2
+		cat pdflatex_$1_$2.log | grep -E "^[!]"
+		exit 1
+	fi
+}
+
+## This function runs biber for a given file.
+## It generates a log file and in case of an error shows the problem in the console.
+## parameter $1 the name of the aux file (without .aux ending)
+## parameter $2 the number of the run (biber requires multiple runs)
+run_biber () {
+	echo -e -n "${GREEN}run $2 biber for $1...${NOCOLOR}"
+	if biber $1 > biber_$1_$2.log ; then
+		echo -e "${GREEN}OK${NOCOLOR}"
+		cat biber_$1_$2.log | grep -E "^(WARN|ERROR)"
+	else
+		echo -e "${RED}fail${NOCOLOR}" >&2
+		cat biber_$1_$2.log | grep -E "^(WARN|ERROR)"
+		exit 1
+	fi
+}
+
+## This function runs bibtex for a given file.
+## It generates a log file and shows the output in the console.
+## parameter $1 the name of the aux file (without .aux ending)
+## parameter $2 the number of the run (bibtex requires multiple runs)
+run_bibtex () {
+	echo -e -n "${GREEN}run $2 bibtex for $1...${NOCOLOR}"
+	if bibtex -terse $1 > bibtex_$1_$2.log ; then
+		echo -e "${GREEN}OK${NOCOLOR}"
+		cat bibtex_$1_$2.log
+	else
+		echo -e "${RED}fail${NOCOLOR}" >&2
+		cat bibtex_$1_$2.log
+		exit 1
+	fi
+}
+
+## This function calls the check_integrity python script and write its output to the console.
+## The script checks for problems within bibtex entries, such as missing fields and wrong field values.
+check_integrity () {
+	echo -e -n "${GREEN}checking bibtex entries...${NOCOLOR}"
+	if python ../scripts/check_integrity.py > check_integrity.log ; then
+		echo -e "${GREEN}OK${NOCOLOR}"
+		cat check_integrity.log
+	else
+		echo -e "${RED}fail${NOCOLOR}" >&2
+		cat check_integrity.log
+		exit 1
+	fi
+}
+
+## This function calls pdflatex with biber
+## parameter $1 the name of the tex file (without .tex ending)
+compile_biber () {
+	run_pdflatex $1 1 "-draftmode"
+	run_biber $1 1
+	run_pdflatex $1 2 "-draftmode"
+	run_biber $1 2
+	run_pdflatex $1 3 ""
+}
+
+## This function calls pdflatex with bibtex
+## parameter $1 the name of the tex file (without .tex ending)
+compile_bibtex () {
+	run_pdflatex $1 1 "-draftmode"
+	run_bibtex $1 1
+	run_pdflatex $1 2 "-draftmode"
+	run_bibtex $1 2
+	run_pdflatex $1 3 ""
+}
+
+## Main script
+change_into_latex_dir
+delete_auxiliary_files
+
+compile_biber short
+compile_biber abrv
+compile_bibtex natbib
+
+check_integrity
```

Comment:
Also generate literature-cleaned? And delete it as aux file, too before testing?

Reply:
added